### PR TITLE
Java11 support - clean version

### PR DESCRIPTION
**Pull request by svanoort**
Clean version of #67 which pulls in the released versions. 

TODO:

- [x] Verify this is safe to run with Java 8 (no issues from the JBoss Marshalling bump)
- [ ] Test upgrading to this version with Pipelines running from old version
- [x] Add Enforcer exclusion for this JBoss Marshalling to allow use of multi-release JAR there.

### CODE REVIEW

**Comment by svanoort** [comment]
Still able to run Pipelines and start Jenkins with Java 8 on core 2.89.x -- as long as we revert the core dependency bump so it does not require 1.129 releases.  

This means the JBoss marshalling dependency is clean - barring enforcer grumbling about the multi-release JAR usage.

File: pom.xml [hunk 1]
```diff
@@ -28,7 +28,7 @@
     <parent>
         <groupId>org.jenkins-ci.plugins</groupId>
         <artifactId>plugin</artifactId>
-        <version>3.12</version>
+        <version>3.16</version>
```
Comment (oleg-nenashev) [inline pom.xml]: Now you can update to 3.17 with newer Animal Sniffer

File: Jenkinsfile [hunk 2]
```diff
@@ -1 +1,2 @@
-buildPlugin()
+//TODO: Currently some tests fail (JENKINS-52014, etc.) 
+buildPlugin(tests: [skip: true])
```
Comment (jglick) [inline Jenkinsfile]: Perhaps better to `@Ignore` those so reviewers can see exactly what is still broken.

File: pom.xml [hunk 3]
```diff
@@ -215,6 +222,25 @@
...
+            <plugin>
+                <artifactId>maven-enforcer-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>display-info</id>
+                        <configuration>
+                            <rules>
+                                <enforceBytecodeVersion>
+                                    <excludes combine.children="append">
+                                        <!-- Prevents enforcer complaining about multi-release JAR -->
```
Comment (jglick) [inline pom.xml]: Where it the upstream bug?

**Comment by svanoort** [comment]
@oleg-nenashev Note: the JBoss marshalling change is *not* back-compatible in the binary format -- that merits a serious upgrade warning (CompatibleSince, etc). 

Does not appear to have a memory leak, after tens of thousands of builds.

Build code, with build rotation set to keep last 100 builds. 

```

class Animal {
    String code;
    
    public Animal(String entry) {
        this.code = entry;
    }
}

for(int i=0; i<5; i++) {
    Animal an = new Animal("named"+i);
}

for (int i=0;i<5; i++) {
    echo 'cheese'
}
```

screen shot 2018-09-21 at 6 39 59 pm

**Comment by oleg-nenashev** [comment]
Deployed the current version as `2.23-20181122.094059-1`

**Comment by oleg-nenashev** [comment]
CC @jenkinsci/java11-support

**Pull request merged**